File: app/src/main/java/com/example/gmailish/data/repository/BlacklistRepository.java
package com.example.gmailish.data.repository;

import com.example.gmailish.data.dao.BlacklistDao;
import com.example.gmailish.data.entity.BlacklistEntity;

import java.util.List;

public class BlacklistRepository {

    private final BlacklistDao blacklistDao;

    public BlacklistRepository(BlacklistDao blacklistDao) {
        this.blacklistDao = blacklistDao;
    }

    // Reads (blocking; add a sync method to DAO if you only had Flow before)
    public List<BlacklistEntity> getAllSync() {
        return blacklistDao.getAllSync();
    }

    public boolean isBlacklisted(String url) {
        return blacklistDao.exists(url);
    }

    // Writes (blocking)
    public void save(BlacklistEntity entry) {
        blacklistDao.upsert(entry);
    }

    public int remove(String url) {
        return blacklistDao.delete(url);
    }
}

File: app/src/main/java/com/example/gmailish/data/repository/BlacklistRepository.kt
package com.example.gmailish.data.repository

import com.example.gmailish.data.dao.BlacklistDao
import com.example.gmailish.data.entity.BlacklistEntity
import kotlinx.coroutines.flow.Flow

class BlacklistRepository(
    private val blacklistDao: BlacklistDao
) {
    // Observe all blacklist entries
    fun observeAll(): Flow<List<BlacklistEntity>> = blacklistDao.observeAll()

    // Check if a URL is blacklisted (one-shot)
    suspend fun isBlacklisted(url: String): Boolean = blacklistDao.exists(url)

    // Add or update a blacklist entry (after server success or local add)
    suspend fun save(entry: BlacklistEntity) = blacklistDao.upsert(entry)

    // Remove a blacklist entry (after server success or local remove)
    suspend fun remove(url: String) = blacklistDao.delete(url)
}